tuf/api/metadata.py

```diff
@@ -1026,6 +1026,10 @@ def __init__(
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         super().__init__(version, spec_version, expires, unrecognized_fields)
+        if any(meta_path == "" for meta_path in meta):
+            raise ValueError("All meta paths must be non-empty strings")
+        if any(meta_path in TOP_LEVEL_ROLE_NAMES for meta_path in meta):
+            raise ValueError("Meta paths must not be top-level metadata roles")
```

Comment:
this came up in the discussion already: meta keys are filenames not rolenames (not necessarily real filenames, but imaginary `{rolename}.json` filenames)

Reply:
You are right.
This begs the question do we want to have a `TOP_LEVEL_FILE_NAMES` set as well?

Reply:
Maybe? It is going to make our "format-independence" even more clearly unrealistic... but we already have ".json" embedded in metadata.py in several places so I don't think this is any worse.

Just or reference, actual format-independence would mean
* probably fixing the spec: meta keys should not be filenames, that makes no sense
* file storage should be tied to serialization (so that when a filename is needed for e.g. saving a file, the serialization implementation gets to decide the final filename: `Serializer.get_filename(rolename: str)` or something)